homeassistant/components/mqtt/config_flow.py

```diff
@@ -469,24 +470,44 @@ async def async_step_broker(
         errors: dict[str, str] = {}
         fields: OrderedDict[Any, Any] = OrderedDict()
         validated_user_input: dict[str, Any] = {}
+        broker_config: dict[str, Any] = {}
+        existing_entry: ConfigEntry | None = None
+        existing_entry_data: MappingProxyType[str, Any] | None = None
+        if self.source == SOURCE_RECONFIGURE:
+            existing_entry = self._get_reconfigure_entry()
+            existing_entry_data = existing_entry.data
         if await async_get_broker_settings(
             self,
             fields,
-            None,
+            existing_entry_data,
             user_input,
             validated_user_input,
             errors,
         ):
+            if existing_entry_data:
+                broker_config.update(
+                    update_password_from_user_input(
+                        existing_entry_data.get(CONF_PASSWORD), validated_user_input
+                    ),
+                )
+            else:
+                broker_config = validated_user_input
+
             can_connect = await self.hass.async_add_executor_job(
                 try_connection,
-                validated_user_input,
+                broker_config,
             )
 
             if can_connect:
+                if existing_entry is not None:
```

Comment:
I think it's better to be explicit:
```suggestion
                if self.source == SOURCE_RECONFIGURE and existing_entry is not None:
```

Reply:
Okay, but note this is already checked

Reply:
Yes I know it's only to satisfy mypy but I think explicit is better for readability.